gelman_rubin.py

```diff
@@ -1,44 +1,73 @@
+written by Dootika Vats and Christina Knudson
+"""
 
 def k_th_batch_mean(k, b, chains):
+    """calculation of a special mean for each chain respectively"""
+
     return (1.0/b) * chains[int((k-1)*b+1):int(k*b),:].sum(axis=0)
 
+
 def combined_estimator_of_mu(m, chains):
+    """The mean of means for different chains"""
+
     return 1.0/m * (chains.mean(axis=1)).sum(axis=0)
 
-def replicated_batch_means_estimator(b, a, m, chains):
+
+def replicated_batch_means_estimator(b, a, m, p, chains):
+    """intermediate step for the following calculation"""
+
     mu = combined_estimator_of_mu(m, chains)
-    s = len(chains[0,0])
-    x = np.zeros((s,s))
+    x = np.zeros((p,p))
+
     for i in range(m):
         for k in range(a):
             k_th_batch_m = k_th_batch_mean(k, b, chains[i])
-            #print(k_th_batch_m)
             x += np.outer((k_th_batch_m - mu),(k_th_batch_m - mu))
-    return (b/(a*m -1))*x
 
-def replicated_lugsail_batch_mean_sestimator(b, a, m, chains):
-    return 2*replicated_batch_means_estimator(float(b), a, m, chains) - replicated_batch_means_estimator(np.floor(float(b/3)), a, m, chains)
+    return (b/(a*m-1)) * x
+
+
+def replicated_lugsail_batch_mean_sestimator(b, a, m, p, chains):
+    """see equation (9) in the paper named at the beginning"""
+
+    return (2 * replicated_batch_means_estimator(float(b), a, m, p, chains)
+           - replicated_batch_means_estimator(np.floor(b/3), a, m, p, chains))
+
 
 #def biased_from_above_estimator(b, a, m, n, chains):
-#    return (float(n-1.0))/n * average_of_the_m_sample_variances(m, n, chains) + (replicated_lugsail_batch_mean_sestimator(b, a, m, chains))/float(n)
-
-def improved_estimator_PSRF(b, a, m, n, chains):
-    s = len(chains[0,0])
-    #print(replicated_batch_means_estimator(b,a,m,chains))
-    #print(np.linalg.det(replicated_lugsail_batch_mean_sestimator(b, a, m, chains)))
-    #print(np.linalg.det(average_of_the_m_sample_variances(m, n, chains)))
-    return ((n-1.)/n+(np.linalg.det(replicated_lugsail_batch_mean_sestimator(b, a, m, chains)/np.linalg.det(average_of_the_m_sample_variances(m, n, chains))))**(1./s)/n)**(1.0/2)
-
-def average_of_the_m_sample_variances(m, n, chains):
-    s= len(chains[0,0])
-    x = np.zeros((s,s))
+
+#    return (float(n-1.0))/n * average_of_the_m_sample_covariances(m, n, chains)
+#           + (replicated_lugsail_batch_mean_sestimator(b, a, m, chains))/float(n)
+
+
+def improved_estimator_PSRF(b, a, m, n, p, chains):
+    """The final revised gelman-rubin statistic receiving b,a as additional
+    parameters for the calculation of the k_th_batch_mean.
+    They are chosen in the way, that n=a*b, whereas n is the chain length
+    (number of samples in one chain). Usual choices for b include round off
+    n^(1/2) or round off n^(1/3) according to the authers.
+    m is the number of chains, n their length, p the dimension of the target
+    distribution and chains the np.ndarray containing the chains in order [m,n,p]
+    """
```

Comment:
```suggestion
    """The final revised Gelman-Rubin statistic receiving b, a as additional
    parameters for the calculation of the k_th_batch_mean.
    They are chosen in the way, that n=a*b. Usual choices for b include round off
    n^(1/2) or round off n^(1/3) according to the authors.
    
    Arguments:
    - m (int): number of chains
    - n (int): length of each chain
    - p (int): dimension of the target distribution 
    - chains (numpy.ndarray): array containing the chains in order [m,n,p]
    """
```
In doc strings, it's often good practice to write an easily readable list of the function arguments, which also indicates their types. There are [standards](https://stackoverflow.com/a/24385103/1656472) for that (my suggestions above conforms to none), such that docstrings can be parsed by software such as [Sphinx](https://www.sphinx-doc.org/en/master/) to automatically generate nicely-formatted documentation.